File: Medium/48_Rotate_image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int temp;
        for(int i=0;i<matrix.size();i++)
        {
            for(int j=0;j<i+1;j++)
            {
                if(i!=j)
                {
                    temp=matrix[i][j];
                    matrix[i][j]=matrix[j][i];
                    matrix[j][i]=temp;
                }
            }
        }
        for(int i=0;i<matrix.size();i++)
        {
            for(int j=0;j<matrix[i].size()/2;j++)
            {
                temp=matrix[i][j];
                matrix[i][j]=matrix[i][matrix[i].size()-1-j];
                matrix[i][matrix[i].size()-1-j]=temp;
            }
        }
        
        

    }
};

File: Easy_python/48_Rotate_image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        temp = []
        for i in range(int(len(matrix)/2)):
            temp = matrix[i]
            matrix[i] = matrix[len(matrix)-i-1]
            matrix[len(matrix)-i-1] = temp
        for i in range(len(matrix)):
            for j in range(i, len(matrix[0])):
                matrix[j][i], matrix[i][j] = matrix[i][j], matrix[j][i]
